Added the CSV Export button In Discharged Patients

AnveshNalimela opened the pull request:
## Proposed Changes

- Fixes #8993 
added export button with restriction based on seven-day period selection.
Introduced an ExportMenu component to handle export of live patients when export is allowed.
Implemented tooltip guidance for when export is not allowed, prompting the user to select a valid seven-day period.

@ohcnetwork/care-fe-code-reviewers


![{CF705BEA-D06D-47A0-A06E-A5C69B52F219}](https://github.com/user-attachments/assets/60be268e-a8f4-49e5-93a2-a51ffa9147f7)


## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Introduced a function to calculate the duration between two dates in days.
	- Added an export feature for discharged patient data, including user notifications and validation.
	- Implemented a new CSV grouping function to enhance data export capabilities.
	- Enhanced localization support with new user-facing strings.
	- Introduced a custom hook for streamlined patient data export.

- **Bug Fixes**
	- Improved validation for phone number entries in the discharged patients list.
	- Enhanced error handling for CSV data processing during exports.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -52,16 +62,96 @@ const DischargedPatientsList = ({
+    last_consultation_medico_legal_case:
+      qParams.last_consultation_medico_legal_case || undefined,
+    last_consultation_encounter_date_before:
+      qParams.last_consultation_encounter_date_before || undefined,
+    last_consultation_encounter_date_after:
+      qParams.last_consultation_encounter_date_after || undefined,
+    last_consultation_discharge_date_before:
+      qParams.last_consultation_discharge_date_before || undefined,
+    last_consultation_discharge_date_after:
+      qParams.last_consultation_discharge_date_after || undefined,
+    last_consultation_admitted_bed_type_list:
+      qParams.last_consultation_admitted_bed_type_list || undefined,
+    last_consultation__consent_types:
+      qParams.last_consultation__consent_types || undefined,
+    last_consultation__new_discharge_reason:
+      qParams.last_consultation__new_discharge_reason || undefined,
+    last_consultation_current_bed__location:
+      qParams.last_consultation_current_bed__location || undefined,
+    number_of_doses: qParams.number_of_doses || undefined,
+    covin_id: qParams.covin_id || undefined,
+    is_kasp: qParams.is_kasp || undefined,
+    is_declared_positive: qParams.is_declared_positive || undefined,
+    last_vaccinated_date_before:
+      qParams.last_vaccinated_date_before || undefined,
+    last_vaccinated_date_after: qParams.last_vaccinated_date_after || undefined,
+    last_consultation_is_telemedicine:
+      qParams.last_consultation_is_telemedicine || undefined,
+    is_antenatal: qParams.is_antenatal || undefined,
+    last_menstruation_start_date_after:
+      (qParams.is_antenatal === "true" &&
+        dayjs().subtract(9, "month").format("YYYY-MM-DD")) ||
+      undefined,
+    ventilator_interface: qParams.ventilator_interface || undefined,
+    diagnoses: qParams.diagnoses || undefined,
+    diagnoses_confirmed: qParams.diagnoses_confirmed || undefined,
+    diagnoses_provisional: qParams.diagnoses_provisional || undefined,
+    diagnoses_unconfirmed: qParams.diagnoses_unconfirmed || undefined,
+    diagnoses_differential: qParams.diagnoses_differential || undefined,
+    review_missed: qParams.review_missed || undefined,
+  };
```
are you sure these filters are present in the discharged patient list?

AnveshNalimela replied:
Review Missed,Facility,Facility type filters are removed

AnveshNalimela replied:
also mesutrational,is_antenatal filters are also removed

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -75,6 +165,88 @@ const DischargedPatientsList = ({
+    lines.map((line: any, i: number) => {
+      const patientId = line.split(",")[0]; // Extract the patient ID from each line
+      if (idsMap.has(patientId)) {
+        idsMap.get(patientId).push(i); // Add the index to the existing array
+      } else {
+        idsMap.set(patientId, [i]); // Create a new array with the current index
+      }
+    });
+
+    const linesWithSameId = Array.from(idsMap.entries())
+      .filter(([_, indices]) => indices.length > 1)
+      .map(([patientId, indices]) => ({
+        patientId,
+        indexSame: indices,
+      }));
+
+    // after getting the array of duplicate patient IDs and there respective linenumbers we will merge the policy IDs of the duplicate patients
+
+    linesWithSameId.map((lineInfo) => {
+      const indexes = lineInfo.indexSame;
+      //get policyid of all the duplicate patients and merge them by seperating them with a semicolon
+      const mergedPolicyId = indexes
+        .map((currentIndex: number) => lines[currentIndex].split(",")[5])
+        .join(";");
+      // replace the policy ID of the first patient with the merged policy ID
+      const arrayOfCurrentLine = lines[indexes[0]].split(",");
+      arrayOfCurrentLine[5] = mergedPolicyId;
+      const lineAfterMerge = arrayOfCurrentLine.join(",");
+      lines[indexes[0]] = `${lineAfterMerge}`;
+    });
+
+    // after merging the policy IDs of the duplicate patients we will remove the duplicate patients from the data
+    const uniqueLines = [];
+    const ids = new Set(); // To keep track of unique patient IDs
+
+    for (const line of lines) {
+      const patientId = line.split(",")[0]; // Extract the patient ID from each line
+      if (!ids.has(patientId)) {
+        uniqueLines.push(line);
+        ids.add(patientId);
```
this is duplicate code, same as present in normal patients list page. do create a utility function out of it and reuse it.

AnveshNalimela replied:
Okay . Created removeDupliactes.ts in utils and reused it

Jacobjeevan commented:
@AnveshNalimela Any updates here?

AnveshNalimela commented:
I will push the requested changes by tmrw

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -52,16 +63,85 @@ const DischargedPatientsList = ({
+      qParams.last_consultation__consent_types || undefined,
+    last_consultation__new_discharge_reason:
+      qParams.last_consultation__new_discharge_reason || undefined,
+    last_consultation_current_bed__location:
+      qParams.last_consultation_current_bed__location || undefined,
```
is this filter even present for discharged patients view?

AnveshNalimela replied:
my Bad i overlooked on that

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -75,6 +155,40 @@ const DischargedPatientsList = ({
+    ],
+  ];
+
+  const calculateDateRangeDuration = (
+    startDate: string,
+    endDate: string,
+  ): number => {
+    if (!startDate && !endDate) return 0;
+    if (startDate && endDate) {
+      return Math.abs(dayjs(endDate).diff(dayjs(startDate), "days"));
+    }
+    return -1; // Invalid range when only one date is set
+  };
```
consider moving this to a utility function. let's not duplicate implementations.

AnveshNalimela replied:
Moved to utils.js and also updated the managePateints.tsx to use ulity function

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,69 @@
+      patientId,
+      indexSame: indices,
+    }));
+
+  // after getting the array of duplicate patient IDs and there respective linenumbers we will merge the policy IDs of the duplicate patients
+
+  linesWithSameId.forEach((lineInfo) => {
+    const indexes = lineInfo.indexSame;
+    if (!indexes.length) return;
+    //get policyid of all the duplicate patients and merge them by seperating them with a semicolon
+    const mergedPolicyId = indexes
+      .map((currentIndex: number) => {
+        const columns = lines[currentIndex].split(",");
+        const policyId = columns[CSV_COLUMNS.POLICY_ID];
+        return policyId?.trim() || "";
+      })
+      .filter(Boolean)
+      .join(";");
+    // replace the policy ID of the first patient with the merged policy ID
+    const arrayOfCurrentLine = lines[indexes[0]].split(",");
+    arrayOfCurrentLine[5] = mergedPolicyId;
+    const lineAfterMerge = arrayOfCurrentLine.join(",");
+    lines[indexes[0]] = `${lineAfterMerge}`;
+  });
+
+  // after merging the policy IDs of the duplicate patients we will remove the duplicate patients from the data
+
+  const cleanedData = Array.from(
+    lines
+      .reduce((map, line) => {
+        const patientId = line.split(",")[CSV_COLUMNS.PATIENT_ID];
+        if (!map.has(patientId)) {
+          map.set(patientId, line);
+        }
+        return map;
+      }, new Map())
+      .values(),
+  ).join("\n"); // Join the unique lines back together
+  return cleanedData;
+};
```
I would recommend making a more generic function that'll group by ID. Let's not make the implementation specific to just to merge Policy ID columns alone alone. Any column could possible have duplicates, which can be merged.

rithviknishad replied:
Also check out [`groupBy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy). Code could be made considerably more readable by using this.

AnveshNalimela replied:
@rithviknishad  i had made the above chnages and pushed . I havent tried doing generic . Would  you complete these PR  by making a more generic function . i had continous exams for couple of weeks . soo

nihal467 commented:
@AnveshNalimela Can you please share the current status of the PR? Otherwise, the PR will be closed in the next 48 hours, and you will be unassigned from the issue due to inactivity.

AnveshNalimela commented:
@nihal467  Due to ongoing exams there is inactivity in PR . I made all the requested changes and few are left to be made. I will make them and push by 12-12-2024

nihal467 commented:
@rithviknishad your review is requested

rithviknishad reviewed (changes requested):
wouldn't this crash the page if any of the thrown Error is encountered?

let's gracefully handle it by showing a notification by try catching the `groupAndMergeByColumn` invocation

AnveshNalimela committed:
Reverted all unwanted file chnages and handled errors with notifications

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

nihal467 committed:
Merge branch 'develop' into issue/#8993/discharge-patients-export

nihal467 commented:
![image](https://github.com/user-attachments/assets/cbcb9571-e2e1-4869-b223-c53e5f86ca81)

- [ ] even after the date range is selected, the export button is not working, @AnveshNalimela make sure your PR is functional before requesting a review

AnveshNalimela commented:
@nihal467 try selecting date range of 7 seven days

nihal467 commented:
> @nihal467 try selecting date range of 7 seven days

check the screenshot

AnveshNalimela committed:
Removed some try-catch which realted to data constiency to make more relaiable

AnveshNalimela committed:
Merge branch 'issue/#8993/discharge-patients-export' of github.com:AnveshNalimela/care_fe into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Chnaged Error to warn

AnveshNalimela committed:
Fixed the issue

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Changed the warn msg to No patients found for export

AnveshNalimela commented:
> > @nihal467 try selecting date range of 7 seven days
> 
> check the screenshot

@nihal The issue is occured due to inconsistent coulumns in data. I fixed that and Whenever there no patients there will error popping up . i changed that error notification to warning

AnveshNalimela committed:
Merge branch 'issue/#8993/discharge-patients-export' of github.com:AnveshNalimela/care_fe into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,81 @@
+export const groupAndMergeByColumn = (
+  data: string,
+  groupByColumnIndex: number,
+  mergeColumnIndices: number[],
```
there's no guarantee the column indice will be same everytime. if we decide to change something in backend (add/remove field), it'd be hard to realise this messed up.

instead use column name.

also i'd recommend passing named args instead of positional args.

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,81 @@
+export const groupAndMergeByColumn = (
```
use proper namings. nobody will know this is a csv utility function without context.

```suggestion
export const csvGroupByColumn = (
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,81 @@
+  mergeColumnIndices: number[],
+  delimiter = ";",
+): string => {
+  if (!data || typeof data !== "string") {
+    throw new Error("Data is in Invalid Format");
+  }
+  const msg = "An error occurred while processing the export";
+
+  if (groupByColumnIndex < 0) {
+    throw new Error(msg);
+  }
+  if (!mergeColumnIndices.length) {
+    throw new Error(msg);
+  }
+  if (mergeColumnIndices.some((index) => index < 0)) {
+    throw new Error(msg);
+  }
```
not required when having named columns instead of index
```suggestion
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,81 @@
+  if (mergeColumnIndices.some((index) => index < 0)) {
+    throw new Error(msg);
+  }
+  const lines = data.split("\n"); // Split the data into individual lines
```
won't the first row be the header? 🤔 

```suggestion
  const [header, ...data] = data.trim().split("\n");
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,81 @@
+    throw new Error(msg);
+  }
+  const lines = data.split("\n"); // Split the data into individual lines
+  if (!lines.length || lines.length === 2) {
+    throw new Error("No  patients found for export");
+  }
+
+  // Remove the last line only if it's empty
+  if (lines[lines.length - 1].trim() === "") {
+    lines.pop();
+  }
```
```suggestion
```

AnveshNalimela committed:
passed columns as args and Updated the function accordingly

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -66,15 +39,14 @@ export const groupAndMergeByColumn = (
-    0, // Group by Patient ID
-    [5], // Merge policy IDs
+    "Patient ID", // Group by Patient ID
+    ["policy ID"], // Merge policy IDs
```
why do we have to specify which all columns to merge? can't we merge for all columns if there are duplicates in any except the primary column (patient id)?

AnveshNalimela replied:
@rithviknishad I restuctured the code without passing the merge columns and merge for columns if have duplicates

AnveshNalimela committed:
changed the warn to error notification in case of any error

AnveshNalimela committed:
Instead of Policy Id merge all columns having duplicates

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,50 @@
+  groupByColumn: string,
+  delimiter = ";",
+): string => {
+  const [header, ...datalines] = data.trim().split("\n"); // Split the data into individual lines
+  const headerColumns = header.split(","); // Extract the header columns
```
code is self explanatory

```suggestion
  const [header, ...datalines] = data.trim().split("\n");
  const headerColumns = header.split(",");
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,50 @@
+      const Index = headerColumns.indexOf(column);
+      const mergedValue = group
+        .map((columns) => columns[Index]?.trim() || "")
+        .filter(Boolean) // Remove empty values
```
```suggestion
        .filter(Boolean)
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,50 @@
+
+    const mergedRow = [...group[0]];
+    headerColumns.forEach((column) => {
+      const Index = headerColumns.indexOf(column);
```
follow naming conventions

```suggestion
      const index = headerColumns.indexOf(column);
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
```diff
@@ -0,0 +1,50 @@
+  return header.concat(mergedLines.join("\n"));
+};
+
+export const preventDuplicatePatients = (data: string): string => {
+  const result = csvGroupByColumn(
+    data,
+    "Patient ID", // Group by Patient ID
+    ";", // Use ";" as the delimiter
+  );
+  return result;
+};
```
let's directly call csvGroupByColumn from manage and discharged patients list instead. 
```suggestion
```

rithviknishad commented on src/Utils/removeDuplicates.ts:
rename the file to something more appropriate. `csvUtils.ts`?

rithviknishad commented on src/components/Patient/ManagePatients.tsx:
```diff
@@ -948,7 +891,16 @@ export const PatientManager = () => {
+                        try {
+                          return preventDuplicatePatients(data);
+                        } catch (e) {
+                          Notification.Warn({
+                            msg: "An unexpected error occurred while processing the export",
+                          });
```
why not show the error message itself?

rithviknishad reviewed (commented):
lgtm, some minor changes

AnveshNalimela committed:
minor chnages and fixed all issues

rithviknishad commented on src/Utils/csvUtils.ts:
```diff
@@ -1,10 +1,10 @@
 export const csvGroupByColumn = (
   data: string,
-  groupByColumn: string,
+  groupByColumn: string = "Patient ID",
```
why does "Patient ID" need to be a default for a generic shared utility function?
```suggestion
  groupByColumn: string,
```

AnveshNalimela replied:
I thought we group them based in primary key (patient ID) by deafult when only data is passed as arg .

rithviknishad replied:
but what if this util is used for assset exports or any other things later?

AnveshNalimela replied:
yeah thats true.

AnveshNalimela committed:
removed patient ID as default parameter

Jacobjeevan commented on src/components/Patient/ManagePatients.tsx:
```diff
@@ -948,7 +891,18 @@ export const PatientManager = () => {
                         return data ?? null;
                       },
-                      parse: preventDuplicatePatientsDuetoPolicyId,
+                      parse: (data) => {
```
Let's switch to using useQuery here.

AnveshNalimela replied:
@Jacobjeevan There is no api request in parsing why to use useQuery

AnveshNalimela replied:
@Jacobjeevan Need to change to all useTanStackQueryInstead to useQuery in both manage patients and dischargedPatientslist . Im holding this PR for long time. Let change them in another issue .

Jacobjeevan replied:
> @Jacobjeevan Need to change to all useTanStackQueryInstead to useQuery in both manage patients and dischargedPatientslist . Im holding this PR for long time. Let change them in another issue .

We are doing slow migration  - switching over whenever a file is modified, rather than opening an issue that switches all at once.

AnveshNalimela replied:
@Jacobjeevan But it is throwing an erro saying tht useQuery cannot  be used an inside a function. Can you confirm where do want me to use useQuery

rithviknishad replied:
https://react.dev/reference/rules/rules-of-hooks

Jacobjeevan replied:
Check the link Rithvik sent; but let us know if you are still having issues with using useQuery.

AnveshNalimela replied:
@Jacobjeevan  Don’t call Hooks inside loops, conditions, nested functions, or try/catch/finally blocks. Couldnt able to use usequery

rithviknishad replied:
google things.... :)

you shouldn't be calling hooks inside functions (other than react components) in the first place. you can achieve it this way:

```jsx
function Component() {
  const { refetch: exportAsCSV } = useQuery({
    queryKey: [...],
    qyeryFn: query(...),
    enabled: false,
  });
  
  
  // in export action:
  action: async () => {
   const data = await exportAsCSV();
   return data;
  },

//
}
```

Jacobjeevan commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -36,9 +36,20 @@ import {
+} from "@/Utils/utils";
+
+import ButtonV2 from "../Common/ButtonV2";
+import ExportMenu from "../Common/Export";
```
absolute imports

Jacobjeevan reviewed (changes requested):
Other than that, looks good.

rithviknishad commented on src/Utils/utils.ts:
```diff
@@ -571,3 +571,14 @@ export const copyToClipboard = async (content: string) => {
   }
 };
+
+export const calculateDateRangeDuration = (
```
duration can be expressed in many ways. since this returns in days, use proper namings.

```suggestion
export const calculateDaysBetween = (
```

nihal467 committed:
Merge branch 'develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
cahnged the calculateDataRange to calculateDaysBetween and changed to absolute imports

AnveshNalimela committed:
Merge branch 'issue/#8993/discharge-patients-export' of github.com:AnveshNalimela/care_fe into issue/#8993/discharge-patients-export

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -293,6 +392,70 @@ const DischargedPatientsList = ({
             />
+            <div className="tooltip w-full md:w-auto" id="patient-export">
+              {!isExportAllowed ? (
+                <ButtonV2
+                  onClick={() => {
+                    advancedFilter.setShow(true);
+                    setTimeout(() => {
+                      const element =
+                        document.getElementById("bed-type-select");
+                      if (element)
+                        element.scrollIntoView({ behavior: "smooth" });
+                      Notification.Warn({
+                        msg: "Please select a seven day period.",
+                      });
+                    }, 500);
+                  }}
+                  className="mr-5 w-full lg:w-fit"
+                >
+                  <CareIcon icon="l-export" />
+                  <span className="lg:my-[3px]">Export</span>
+                </ButtonV2>
```
We no longer use `ButtonV2` in newer codebases. Switch to using Button from shadcn

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -136,6 +212,29 @@ const DischargedPatientsList = ({
+    ],
+  ];
+
+  const durations = date_range_fields.map(([startDate, endDate]) =>
+    calculateDaysBetween(startDate, endDate),
+  );
+
+  const isExportAllowed =
+    durations.every((x) => x >= 0 && x <= 7) &&
+    !durations.every((x) => x === 0);
```
this logic can be simplified by using `Array.some()` instead.

```
  const isExportAllowed = dateRangeFields.some(...);
```

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -136,6 +212,29 @@ const DischargedPatientsList = ({
     }
   }, [qParams]);
 
+  const date_range_fields = [
```
use proper naming conventions

https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#variable_names

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Replaced buttonv2 with button,updated duration logic,dateRangeFields variable updated

Jacobjeevan commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -136,6 +213,26 @@ const DischargedPatientsList = ({
+
+  const isExportAllowed = dateRangeFields.some(([startDate, endDate]) => {
+    const days = calculateDaysBetween(startDate, endDate);
+    return days >= 0 && days <= 7 && days !== 0;
```
why check for greater than equal to zero and then for not equal to zero here? 🤔  Greater than zero alone should suffice.

Jacobjeevan commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -293,6 +390,72 @@ const DischargedPatientsList = ({
+                      if (element)
+                        element.scrollIntoView({ behavior: "smooth" });
+                      Notification.Warn({
+                        msg: "Please select a seven day period.",
```
Let's add this to translation while we are at it.

AnveshNalimela committed:
Merge branch 'develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
removed >=0 and added translations for export and seven day period

AnveshNalimela committed:
Replaced with useQuery

AnveshNalimela committed:
Removed console logs

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -233,6 +234,24 @@ const DischargedPatientsList = ({
+  const pathParams = { facility_external_id };
+
+  const { refetch: exportAsCSV } = useQuery({
+    queryKey: ["Discharged Patients", queryParams],
+    queryFn: async () => {
+      const data = await query(routes.listFacilityDischargedPatients, {
+        queryParams,
+        pathParams,
+      });
+      return data;
+    },
+    enabled: false,
+  });
```
refer docs... query returns a callback not the data. also you are not passing the abort controller this way.

also dependencies need not be listed, since it's always manually trigerred. you can simplify things here.

```suggestion
    queryKey: ["discharged-patients-csv"],
    queryFn: query(routes.listFacilityDischargedPatients, {
        queryParams: { ...params, csv: true },
        pathParams: { facility_external_id },
      }),
    enabled: false,
  });
```

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -233,6 +234,24 @@ const DischargedPatientsList = ({
     return days > 0 && days <= 7 && days !== 0;
   });
 
+  const queryParams = {
+    ...params,
+    csv: true,
+  };
+  const pathParams = { facility_external_id };
+
```
```suggestion
```

rithviknishad commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -419,20 +438,18 @@ const DischargedPatientsList = ({
+                        });
+                        console.log(data);
+
+                        return data || null;
```
what is this?

also you shouldn't be getting the `isError` fields and all. you should get the data directly. error handling is done by the global query client.

why call `data` and pass abort controller here?

AnveshNalimela committed:
simplified the useQuery implemnenation

rithviknishad commented on src/components/Patient/ManagePatients.tsx:
```diff
@@ -221,74 +226,20 @@ export const PatientManager = () => {
+      queryParams: { ...params, csv: true, facility: qParams.facility },
+    }),
+    enabled: false,
+  });
```
let's do this

AnveshNalimela committed:
Added usePatientExport.ts

AnveshNalimela committed:
Removed redudant logic in isExportAllowed

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

Jacobjeevan commented on src/components/Facility/DischargedPatientsList.tsx:
```diff
@@ -136,6 +213,32 @@ const DischargedPatientsList = ({
+    { facility_external_id },
+  );
+
   const { data: districtData } = useTanStackQueryInstead(routes.getDistrict, {
```
Few more of these - switch to useQuery.

Other than that, looks good.

Currently doesn't support filtering by discharge date range (BE blocker; will need to open an issue there).

AnveshNalimela replied:
@Jacobjeevan  Replaced all queries using useQuery Hook .And Should I remove the filtering by discharge date or let it be

AnveshNalimela replied:
@Jacobjeevan Could you check the changes made and update the label

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Replaced all usetanstackqueryinstaed with useQuery

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

AnveshNalimela committed:
Merge branch 'ohcnetwork:develop' into issue/#8993/discharge-patients-export

rithviknishad commented:
no longer relevant in new architecture
